BossMod/Modules/Endwalker/NormalTrials/Trial7Zeromus/Trial7ZeromusStates.cs

```diff
@@ -0,0 +1,302 @@
+namespace BossMod.Endwalker.NormalTrials.Trial7Zeromus
+{
+    class Trial7ZeromusStates : StateMachineBuilder
+    {
+        public Trial7ZeromusStates(BossModule module) : base(module)
+        {
+            SimplePhase(0, Phase1, "P1")
+                .Raw.Update = () => Module.PrimaryActor.IsDestroyed || Module.PrimaryActor.IsDead || Module.PrimaryActor.HP.Cur <= 1 || (Module.PrimaryActor.CastInfo?.IsSpell(AID.RendTheRift) ?? false);
+            DeathPhase(1, Phase2); // starts at around 25%
+        }
+
+        private void Phase1(uint id)
+        {
+            AbyssalNoxEchoesSableThread(id, 6.2f, false);
+            DarkMatter(id + 0x10000, 7.4f);
+            VisceralWhirl(id + 0x20000, 5.3f);
+            Flare(id + 0x30000, 7.1f);
+            VoidBioVisceralWhirl(id + 0x40000, 5.1f);
+            BigBang(id + 0x50000, 5.1f);
+            VoidMeteor(id + 0x60000, 10.2f);
+            VisceralWhirlChainsBombs(id + 0x70000, 4.4f);
+            DarkMatterForkedLightningDarkBeckons(id + 0x80000, 2.4f);
+            BlackHole(id + 0x90000, 5.3f);
+            BigCrunch(id + 0xA0000, 4.5f);
+            AbyssalNoxEchoesSableThread(id + 0xB0000, 10.3f, true);
+            SparklingBrandingFlare(id + 0xC0000, 7.5f);
+            VoidBioVisceralWhirl(id + 0xD0000, 3.9f);
+            DarkMatter(id + 0xE0000, 7.1f);
+            SparklingBrandingFlare(id + 0xF0000, 5.2f);
+
+            SimpleState(id + 0xFF0000, 10, "???");
+        }
+
+        private void Phase2(uint id)
+        {
+            RendTheRift(id, 0);
+            DimensionalSurgeNostalgia(id + 0x10000, 8.2f);
+            FlowOfTheAbyss(id + 0x20000, 7.4f, false);
+            FlowOfTheAbyss(id + 0x30000, 10.7f, true);
+            DimensionalSurgeNostalgia(id + 0x40000, 7.7f);
+            FlowOfTheAbyss(id + 0x50000, 7.4f, true);
+            FlowOfTheAbyss(id + 0x60000, 10.7f, true);
+            DimensionalSurgeNostalgia(id + 0x70000, 7.7f);
+
+            SimpleState(id + 0xFF0000, 10, "???");
+        }
+
+        private void AbyssalNoxEchoesSableThread(uint id, float delay, bool second)
+        {
+            Cast(id, AID.AbyssalNox, delay, 5);
+            ComponentCondition<AbyssalEchoes>(id + 0x1000, 0.1f, comp => comp.Casters.Count > 0)
+                .ActivateOnEnter<AbyssalEchoes>()
+                .ExecOnEnter<AbyssalEchoes>(comp => comp.MaxCasts = 0); // don't show aoes before doom, this is misleading
+            ComponentCondition<AbyssalEchoes>(id + 0x1010, 5, comp => comp.Casters.Count > 5, "1 hp"); // dooms are slightly staggered apply around here
+            ComponentCondition<AbyssalEchoes>(id + 0x1020, 11, comp => comp.NumCasts > 0, "Circles 1")
+                .ExecOnEnter<AbyssalEchoes>(comp => comp.MaxCasts = 5);
+            ComponentCondition<AbyssalEchoes>(id + 0x1030, 5, comp => comp.Casters.Count == 0, "Circles 2")
```

Comment:
Ok, I've just watched a video of a normal version of the trial - for this very first mechanic I don't see the two staggered sets of aoes with 5s between them, like extreme has.

So I'm confused why is this a thing here.